synapse/config/cache.py

```diff
@@ -217,9 +221,16 @@ def read_config(self, config, **kwargs) -> None:
                     e.message  # noqa: B306, DependencyException.message is a property
                 )
 
-        expiry_time = cache_config.get("expiry_time")
-        if expiry_time:
-            self.expiry_time_msec: Optional[int] = self.parse_duration(expiry_time)
+        expire_caches = cache_config.get("expire_caches", True)
+        cache_entry_ttl = cache_config.get("cache_entry_ttl")
+
+        if expire_caches:
+            if cache_entry_ttl:
+                self.expiry_time_msec: Optional[int] = self.parse_duration(
+                    cache_entry_ttl
+                )
+            else:
+                self.expiry_time_msec = 1800000
```

Comment:
May be worth adding a quick comment to indicate that this equates to 30 minutes.